File: src/main/java/com/mikuac/yuri/utils/BeanUtils.java
package com.mikuac.yuri.utils;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

/**
 * @author zero
 */
@Component
public class BeanUtils implements ApplicationContextAware {

    private static ApplicationContext applicationContext = null;

    @Override
    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
        if (BeanUtils.applicationContext == null) {
            BeanUtils.applicationContext = applicationContext;
        }
    }

    public static ApplicationContext getApplicationContext() {
        return applicationContext;
    }

    public static Object getBean(String beanName) {
        return applicationContext.getBean(beanName);
    }

    public static <T> T getBean(Class<T> c) {
        return applicationContext.getBean(c);
    }

    public static <T> T getBean(String name, Class<T> c) {
        return getApplicationContext().getBean(name, c);
    }

}

File: src/main/kotlin/com/mikuac/yuri/utils/BeanUtils.kt
package com.mikuac.yuri.utils

import org.springframework.beans.BeansException
import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.stereotype.Component

@Component
class BeanUtils : ApplicationContextAware {

    companion object {
        private var ctx: ApplicationContext? = null

        fun getBean(beanName: String): Any {
            return ctx!!.getBean(beanName)
        }

        fun <T> getBean(c: Class<T>): T {
            return ctx!!.getBean(c)
        }

        fun <T> getBean(name: String, c: Class<T>): T {
            return ctx!!.getBean(name, c)
        }

        fun getCtx(): ApplicationContext {
            return ctx!!
        }
    }

    @Throws(BeansException::class)
    override fun setApplicationContext(applicationContext: ApplicationContext) {
        if (ctx == null) {
            ctx = applicationContext
        }
    }

}